File: src/main/java/com/fixtab/app/models/db/activities/ActivityModel.java
package com.fixtab.app.models.db.activities;

import jakarta.persistence.*;
import lombok.*;

import java.util.Date;

import com.fixtab.app.models.db.BaseEntity;
import com.fixtab.app.models.db.employees.EmployeeModel;

@Setter
@Getter
@NoArgsConstructor
@AllArgsConstructor
@Builder
@Table(name = "activities")
@Entity
public class ActivityModel extends BaseEntity {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "activityid")
    private int activityId;

    @Column(name = "sequencenumber")
    private int sequenceNumber;

    @Column(name = "description")
    private String description;

    @Column(name = "cancelled")
    private boolean cancelled;

    @Column(name = "status")
    private int status;

    @Column(name = "createdate")
    private Date createDate;

    @Column(name = "statusupatedate")
    private Date statusUpateDate;

     /*
    * FOREIGN KEYS !!!
    * */
    @OneToOne(cascade = CascadeType.ALL)
    @JoinColumn(name = "activitytypeid")
    private ActivityTypeModel activityType;

    @ManyToOne(cascade = CascadeType.ALL)
    @JoinColumn(name = "employeeid")
    private EmployeeModel employee;

    @OneToOne(cascade = CascadeType.ALL)
    @JoinColumn(name = "resultid")
    private ResultDictionaryModel result;
}

File: client-application/src/app/entitites/activity.model.ts
import { ActivityType } from "./activity-type.model";
import { Employee } from "./employee-model";
import { StatusDictionary } from "./result-dictionary.model";
import { Status } from "./result.model";

interface IActivity {
  activityId?: number;
  sequenceNumber?: number;
  description?: string;
  cancelled?: boolean;
  result?: string;
  createDate?: string;
  statusUpateDate?: string;
  activityType?: ActivityType;
  manager?: Employee;
  employee?: Employee;
  status?: Status
}

interface IActivityRequest {
  sequenceNumber?: number;
  description?: string;
  activityType?: ActivityType;
  employeeId?: number;
}

export class Activity implements IActivity {
  constructor(
  public activityId?: number,
  public sequenceNumber?: number,
  public description?: string,
  public cancelled?: boolean,
  public result?: string,
  public createDate?: string,
  public statusUpateDate?: string,
  public activityType?: ActivityType,
  public manager?: Employee,
  public employee?: Employee,
  public status?: Status
  ) {}
}

export class ActivityRequest implements IActivityRequest {
  constructor(
  public sequenceNumber?: number,
  public description?: string,
  public activityType?: ActivityType,
  public employeeId?: number
  ) {}
}
